test/e2e/multihoming.go

```diff
@@ -1194,3 +1242,66 @@ func kickstartPod(cs clientset.Interface, configuration podConfiguration) *v1.Po
 	}, 2*time.Minute, 6*time.Second).Should(Equal(v1.PodRunning))
 	return createdPod
 }
+
+func createNad(namespace string, nadClient nadclient.K8sCniCncfIoV1Interface, netConfig networkAttachmentConfig) {
+	By("creating the attachment configuration")
+	_, err := nadClient.NetworkAttachmentDefinitions(namespace).Create(
+		context.Background(),
+		generateNAD(netConfig),
+		metav1.CreateOptions{},
+	)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func setupLocalnetUnderlay(cs clientset.Interface, netConfig networkAttachmentConfig) func() {
```

Comment:
I would rather have the setup section in the context's `BeforeEach` and the teardown in the `AfterEach` .

Plus, return errors and assert them in the test's body / setup / teardown. The test becomes IMO more readable.